x-pack/plugins/observability_solution/infra/server/lib/alerting/metric_threshold/lib/create_bucket_selector.ts

```diff
@@ -49,7 +45,7 @@ export const createBucketSelector = (
           },
           script: createConditionScript(
             condition.warningThreshold as number[],
-            condition.warningComparator as Comparator
+            condition.warningComparator as COMPARATORS
```

Comment:
No, updated.

Reply:
I see in the type, this might be undefined:

![image](https://github.com/elastic/kibana/assets/12370520/d9a348f0-0e70-4528-b349-e5f5003a8e9e)

But since we are checking this in hasWarn:
```
  const hasWarn = condition.warningThreshold != null && condition.warningComparator != null;
```

We can change it to `condition.warningComparator!` as we know it will not be undefined in this case. Not sure why typescript does not recognize it itself:

![image](https://github.com/elastic/kibana/assets/12370520/b744bc3b-5faa-42b3-9c8b-528c29adc46b)